.github/workflows/build-and-deploy.yaml

```diff
@@ -0,0 +1,129 @@
+name: build-and-deploy
+
+on:
+  push:
+    branches:
+      - main
+permissions: write-all
```

Comment:
Since you are specifying permissions on per-job basis this statement is redundant

Reply:
Updated